src/processing/ParameterizedOperationsUtil.java

```diff
@@ -0,0 +1,197 @@
+package processing;
+
+import java.util.*;
+import processing.boardobject.*;
+import processing.utility.*;
+
+/**
+ *
+ * @author Devansh Singh Rathore
+ * @reviewer Shruti Umat
+ */
+
+/**
+ * Class for implementing selected(non-erase, non-reset) object's Color-change and Rotate
+ * operation.
+ */
+public class ParameterizedOperationsUtil {
+
+    /**
+     * Function to update undo-redo stack. Undo-Redo module cannot call this function inorder to
+     * avoid logical errors.
+     */
+    private static void stackUtil(BoardObject newObj) {
+
+        UndoRedo.pushIntoStack(newObj);
+        return;
+    }
+
+    /**
+     * Function to calculate the centre of object's pixel by averaging the x and y coordinates of
+     * all the pixels in the given object.
+     */
+    private static Position findCentre(ArrayList<Pixel>pixels) {
+
+        Position centrePos = new Position(0, 0);
+        int numOfPixels = pixels.size();
+        for(int i = 0; i < numOfPixels; i++) {
+
+            centrePos.r += pixels.get(i).position.r;
+            centrePos.c += pixels.get(i).position.c;
+        }
+
+        // averaging, followed by Narrowing Type Casting
+        centrePos.r = (int)(centrePos.r / numOfPixels);
+        centrePos.c = (int)(centrePos.c / numOfPixels);
+
+        return centrePos;
+    }
+
+    /**
+     * Function calculates the rotation matrix.
+     */
+    private static double[][] rotationMatrix(Angle angleOfRotation) {
```

Comment:
Type names (Enum, Class, Struct) are named as nouns, and Method names (functions) as verbs. So "findCentre" is a good name where as "rotationMatrix" is not. I would rather call it "computeRotationMatrix". JFYI.

Reply:
I'll change it